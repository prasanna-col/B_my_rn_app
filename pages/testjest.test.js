import React from 'react';
import {cleanup, render} from '@testing-library/react-native';
import renderer, { create } from 'react-test-renderer'
import TestJT from './testjest';

afterEach(cleanup);

describe('TestJT', () => {

  test("function test testID",()=>{
    let FuncTest = create(<TestJT/>).getInstance();
    expect(FuncTest.dataFunc(5)).toBe(25)
  })

  it('should show hello world', () => {
    const helloWorldText = 'Hello World!';
    const notFoundText = 'Not found text';

    const {toJSON, getByText, queryByText} = render(<TestJT />);

    const foundHelloWorldTextElement = getByText(helloWorldText);
    const notFoundTextElement = queryByText(notFoundText);

    expect(foundHelloWorldTextElement.props.children).toEqual(helloWorldText);
    expect(notFoundTextElement).toBeNull();
    expect(toJSON()).toMatchSnapshot();
  });

  it('should find the button via testId', () => {
    const testIdName = 'pressMeButton';

    const {getByTestId} = render(<TestJT />);

    const foundButton = getByTestId(testIdName);

    expect(foundButton).toBeTruthy();
  });

  it('should find the button via accessibilityLabel', () => {
    const accessibilityLabel = 'Press me';

    const {getByA11yLabel} = render(<TestJT />);

    const foundButton = getByA11yLabel(accessibilityLabel);

    expect(foundButton).toBeTruthy();
  });

  it('should find the button title', () => {
    const title = 'Press me!';

    const {getByText} = render(<TestJT />);

    const foundButtonTitle = getByText(title);

    expect(foundButtonTitle.props.children).toEqual(title);
  });
});