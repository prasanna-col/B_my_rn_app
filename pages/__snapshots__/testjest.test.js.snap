// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestJT should show hello world 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
    }
  }
>
  <View
    style={
      Object {
        "marginTop": 32,
        "paddingHorizontal": 24,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 24,
          "fontWeight": "600",
        }
      }
    >
      Hello World!
    </Text>
  </View>
  <View
    accessibilityLabel="Press me"
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="pressMeButton"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press me!
      </Text>
    </View>
  </View>
</View>
`;
